### PR TITLE
Add who started poll and when

### PR DESCRIPTION
**Pull request by Lord-Haji**

### CODE REVIEW

File: chat-plugins/poll.js [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 'use strict';
 
+const moment = require('moment');
```
Comment (Zarel) [inline chat-plugins/poll.js]: 1. you can't do this without adding this to package.json

2. we already have a duration library in Tools, no reason to bring another in

File: chat-plugins/poll.js [hunk 2]
```diff
@@ -62,14 +65,15 @@ class Poll {
 		this.options.forEach((option, number) => {
 			output += '<div style="margin-top: 5px"><button value="/poll vote ' + number + '" name="send" title="Vote for ' + number + '. ' + Tools.escapeHTML(option.name) + '">' + number + '. <strong>' + Tools.escapeHTML(option.name) + '</strong></button></div>';
 		});
-		output += '<div style="margin-top: 7px; padding-left: 12px"><button value="/poll results" name="send" title="View results - you will not be able to vote after viewing results"><small>(View results)</small></button></div>';
+		output += '<div style="margin-top: 7px; padding-left: 12px"><hr><button value="/poll results" name="send" title="View results - you will not be able to vote after viewing results"><small>(View results)</small></button></div>';
```
Comment (Zarel) [inline chat-plugins/poll.js]: Would an `<hr>` even look good here?

Comment (Zarel) [inline chat-plugins/poll.js]: What does this look like?

File: chat-plugins/poll.js [hunk 3]
```diff
@@ -62,14 +65,15 @@ class Poll {
...
 		output += '</div>';
 
 		return output;
 	}
 
 	generateResults(ended, option) {
 		let icon = '<span style="border:1px solid #' + (ended ? '777;color:#555' : '6A6;color:#484') + ';border-radius:4px;padding:0 3px"><i class="fa fa-bar-chart"></i> ' + (ended ? "Poll ended" : "Poll") + '</span>';
+		let totalVotes = '<p align="left">[Total Votes: ' + this.totalVotes + '] <i>(Started by ' + this.startedUser + ' ' + moment(this.startTime).fromNow() + '.)</i></p>';
```
Comment (Zarel) [inline chat-plugins/poll.js]: This isn't actually used anywhere (I'm assuming that's what Travis is complaining about)

File: chat-plugins/poll.js (hunk 2 above)
Comment (Lord-Haji) [inline chat-plugins/poll.js]: it would kinda give space
and a line to separate the poll options and View Results Button

Comment (Lord-Haji) [inline chat-plugins/poll.js]: :tada:

Comment (kotarou3) [inline chat-plugins/poll.js]: Screenshot?

File: chat-plugins/poll.js (hunk 3 above)
Comment (Lord-Haji) [inline chat-plugins/poll.js]: lol me 2

Comment (Lord-Haji) [inline chat-plugins/poll.js]: wait I found wut travis is complaining about

File: chat-plugins/poll.js (hunk 1 above)
Comment (Lord-Haji) [inline chat-plugins/poll.js]: it will separate to another dependency make it arranged in a compact way

Comment (Lord-Haji) [inline chat-plugins/poll.js]: No we can
but it would make inconvinient to newbies

**Comment by panpawn** [comment]
lol, copying my code

**Comment by panpawn** [comment]
https://github.com/panpawn/Pokemon-Showdown/commit/5840eefb78bd07f48e9eea6634c373b21a5c9f7f

you didn't copy and paste right, either

**Comment by Lord-Haji** [comment]
nah it isnt
i used moment coz Tolls.duration fails

**Comment by Zarel** [comment]
Except you used moment wrong. Without actually adding moment to package.json, this code will crash.

And you'd be better off figuring out why it failed than bringing in an outside dependency. It might be because you spelled `Tools` wrong.

File: chat-plugins/poll.js [hunk 4]
```diff
@@ -20,6 +22,7 @@ class Poll {
 		this.totalVotes = 0;
 		this.timeout = null;
 		this.timeoutMins = 0;
+		this.startedUser = user.name;
```
Comment (panpawn) [inline chat-plugins/poll.js]: ``user`` is not defined here

File: chat-plugins/poll.js (hunk 3 above)
Comment (panpawn) [inline chat-plugins/poll.js]: ``this.startTime`` is also not defined here

**Pull request closed without merging**